src/PIL/JpegXlImagePlugin.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import struct
+from io import BytesIO
+
+from . import Image, ImageFile
+
+try:
+    from . import _jpegxl
+
+    SUPPORTED = True
+except ImportError:
+    SUPPORTED = False
+
+
+## Future idea:
+## it's not known how many frames does animated image have
+## by default, _jxl_decoder_new will iterate over all frames without decoding them
+## then libjxl decoder is rewinded and we're ready to decode frame by frame
+## if OPEN_COUNTS_FRAMES is False, n_frames will be None until the last frame is decoded
+## it only applies to animated jpeg xl images
+# OPEN_COUNTS_FRAMES = True
+
+
+def _accept(prefix: bytes) -> bool:
+    is_jxl = prefix.startswith(
+        (b"\xff\x0a", b"\x00\x00\x00\x0c\x4a\x58\x4c\x20\x0d\x0a\x87\x0a")
+    )
+    if is_jxl and not SUPPORTED:
+        msg = "image file could not be identified because JXL support not installed"
+        raise SyntaxError(msg)
+    return is_jxl
+
+
+class JpegXlImageFile(ImageFile.ImageFile):
+    format = "JPEG XL"
+    format_description = "JPEG XL image"
+    __loaded = 0
+    __logical_frame = 0
+
+    def _open(self) -> None:
+        self._decoder = _jpegxl.JpegXlDecoder(self.fp.read())
+
+        (
+            self._size,
+            self._mode,
+            self.is_animated,
+            tps_num,
+            tps_denom,
+            self.info["loop"],
+            n_frames,
+        ) = self._decoder.get_info()
+
+        self._tps_dur_secs = 1
+        self.n_frames: int | None = 1
+        if self.is_animated:
+            self.n_frames = None
+            if n_frames > 0:
+                self.n_frames = n_frames
+                self._tps_dur_secs = tps_num / tps_denom
```

Comment:
I'm guessing that `n_frames` is zero when there is an animated file but one of the frames can't be loaded? As in, a well-formed image shouldn't cause this, but it is a file where we will never know the number of frames even when seeking forward, because it will fail.